web/src/main/webapp/lamp.ts

```diff
@@ -0,0 +1,412 @@
+        this.updateControlPointsView();
+        this.updateHeatMapView();
+    }
+
+    public updateRemoteTable(table: RemoteTableObject) {
+        this.lampTableObject = table;
+    }
+
+    public updateControlPoints(pointSet: PointSet2D) {
+        this.controlPoints = pointSet;
+        /* Set the coordinate system of the plot */
+        let [minX, maxX] = [Math.min(...this.controlPoints.points.map(p => p.x)), Math.max(...this.controlPoints.points.map(p => p.x))];
+        let [minY, maxY] = [Math.min(...this.controlPoints.points.map(p => p.y)), Math.max(...this.controlPoints.points.map(p => p.y))];
+        let maxRange = Math.max(maxX - minX, maxY - minY);
+        this.minX = minX;
+        this.minY = minY;
+        this.maxX = minX + maxRange;
+        this.maxY = minY + maxRange;
+
+        this.updateControlPointsView();
+        this.applyLAMP();
+    }
+
+    public updateHeatMap(heatMap: HeatMapData) {
+        this.heatMap = heatMap;
+        this.updateHeatMapView()
+    }
+    public updateHeatMapView() {
+        if (this.heatMap == null)
+            return;
+        this.heatMapChart.selectAll("*").remove();
+        this.heatMapChart.append("rect")
+            .attr("x", 0)
+            .attr("y", 0)
+            .attr("width", this.heatMapChart.attr("width"))
+            .attr("height", this.heatMapChart.attr("height"))
+            .style("fill", "none")
+            .style("stroke", "black");
+        let pointWidth = this.heatMapChart.attr("width") / this.heatMap.buckets.length;
+        let pointHeight = this.heatMapChart.attr("height") / this.heatMap.buckets[0].length;
+        let dots = [];
+        let max = 0;
+        for (let i = 0; i < this.heatMap.buckets.length; i++) {
+            for (let j = 0; j < this.heatMap.buckets[0].length; j++) {
+                let val = this.heatMap.buckets[i][j];
+                if (val > 0) {
+                    dots.push({x: i * pointWidth, y: this.heatMapChart.attr("height") - (j + 1) * pointHeight, v: val});
+                    max = Math.max(val, max);
+                }
+            }
+        }
+        this.colorMap.min = 1;
+        this.colorMap.max = max;
+        this.colorLegend.redraw();
+        this.heatMapChart.selectAll()
+            .data(dots)
+            .enter()
+            .append("rect")
+            .attr("x", d => d.x)
+            .attr("y", d => d.y)
+            .attr("data-val", d => d.v)
+            .attr("width", pointWidth)
+            .attr("height", pointHeight)
+            .style("stroke-width", 0)
+            .style("fill", d => this.colorMap.apply(d.v));
+    }
+
+    public applyLAMP() {
+        let xBuckets = Math.ceil(this.heatMapChart.attr("width") / Resolution.minDotSize);
+        let yBuckets = Math.ceil(this.heatMapChart.attr("height") / Resolution.minDotSize);
+        let xColAndRange = {
+            min: this.minX,
+            max: this.maxX,
+            samplingRate: 1.0,
+            columnName: "LAMP1",
```

Comment:
What if this column already exists?
Maybe we should add a method that generates a fresh column name from a schema and a suggested prefix.